### PR TITLE
Fixes 24969. Fix eds endpoint selection for subsets with no or empty …

### PR DESCRIPTION
**Pull request by robertpanzer**
…subset labels

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

This PR fixes the behavior reported in #24969.
Currently, when a DestinationRule has a subset with no label selector it is treated as an array of label.Instance with one element, `nil`.

When computing the endpoints for EDS in `DiscoveryServer.loadAssignmentsForClusterIsolated()` that match this subset there is a special handling for label selectors with no elements which results in all endpoints accepted:
https://github.com/istio/istio/blob/2fa151ab385999535517c489165164a8deb650b1/pkg/config/labels/collection.go#L23-L37

But when c contains a single element `nil`, then the function always returns false when the endpoint also has no label, resulting in no endpoints assigned to that cluster.
(I didn't test it, but I would assume that if an endpoint has at least one label that the code should panic in line 32 on a nil dereference)

The fix handles no or an empty label selector on the subset as `nil` instead of of `[]labels.Instance{nil}`

Note: I already opened #24970 which targeted the release-1.6 branch. 
This PR is for master which seems to show the same problem.

### CODE REVIEW

**Comment by robertpanzer** [comment]
Thanks for merging!
It would be great if this could make it into 1.6 too.

**Comment by ramaraochavali** [comment]
I added the cherrypick label to push it to 1.6. But nothing happened. Let me try again

**Comment by ramaraochavali** [comment]
/cherrypick/release-1.6

**Comment by ramaraochavali** [comment]
@howardjohn @rshriram cherrypick seems to be not working for me. Can u please cherrypick this for 1.6?
